fahrschein-http-jdk11/src/main/java/org/zalando/fahrschein/http/jdk11/JavaNetRequestFactory.java

```diff
@@ -0,0 +1,33 @@
+package org.zalando.fahrschein.http.jdk11;
+
+import org.zalando.fahrschein.http.api.ContentEncoding;
+import org.zalando.fahrschein.http.api.Request;
+import org.zalando.fahrschein.http.api.RequestFactory;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.time.Duration;
+
+public final class JavaNetRequestFactory implements RequestFactory {
+
+    private final HttpClient client;
+    private final Duration timeout;
+    private final ContentEncoding contentEncoding;
+
+    /**
+     * @param client the HTTP client
+     * @param timeout the timeout duration. See {@code java.net.http.HttpRequest.Builder#timeout}
+     * @param contentEncoding the encoding for publishing events
+     */
+    public JavaNetRequestFactory(HttpClient client, Duration timeout, ContentEncoding contentEncoding) {
```

Comment:
don't be sorry, but what is the reason of not setting a timeout?

Reply:
Because `java.net.http.HttpRequest.Builder` does not require one.

Reply:
I am assuming that for our use-case (server-to-server comm) it is advisable to always have a timeout set.... But maybe this is assuming too much. Can also make it optional.

Reply:
It may be advisable, but we should not force the user to set one when the related API does not do so.

For example, in our other implementations such as `HttpComponentsRequestFactoryTest` and `SimpleRequestFactory` we do not require a timeout.

Reply:
👍  will make it optional